pkg/tools/gogen/gogen.go

```diff
@@ -0,0 +1,101 @@
+package gogen
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"strings"
+
+	"kusionstack.io/kclvm-go/pkg/logger"
+	"kusionstack.io/kclvm-go/pkg/spec/gpyrpc"
+)
+
+type GenStruct map[StructName][]*Field
+
+type StructName string
+
+type Field struct {
+	Name       string
+	SchemaType string
+	OmitEmpty  bool
+	Tag        string
+}
+
+func parseKclType(ktList []*gpyrpc.KclType) *GenStruct {
+	var schemaName StructName
+	genStruct := make(GenStruct, 0)
+	for _, kt := range ktList {
+		if kt.Type == "schema" {
+			schemaName = StructName(kt.SchemaName)
+			var schemaList []*Field
+			for k, p := range kt.Properties {
+				var field Field
+				field.Name = strings.ToUpper(k[:1]) + k[1:]
+				switch p.Type {
+				case "schema":
+					field.SchemaType = "*" + p.SchemaName
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "schema")
+				case "dict":
+					kType := p.Key.Type
+					if kType == "str" {
+						kType = "string"
+					} else if kType == "int" {
+						kType = "int"
+					}
+					vType := p.Item.Type
+					if vType == "schema" {
+						vType = p.Item.SchemaName
+					}
+					field.SchemaType = "map[" + kType + "]" + vType
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "{"+p.Key.Type+":"+vType+"}")
+				case "list":
+					vType := p.Item.Type
+					if vType == "schema" {
+						vType = p.Item.SchemaName
+					}
+					field.SchemaType = "[]" + "*" + vType
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "["+vType+"]")
+				case "str":
+					field.SchemaType = "string"
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "str")
+				case "int":
+					field.SchemaType = "int"
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "int")
+				case "float":
+					field.SchemaType = "float32"
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "float")
+				case "bool":
+					field.SchemaType = "bool"
+					field.Tag = fmt.Sprintf(`kcl:"name=%s,type=%s"`, k, "bool")
+				case "null":
+					field.SchemaType = "nil"
```

Comment:
nil type or any type or `interface{}`?

Reply:
I don't know this usage about kcl schema 
```
schema Movie:
	desc: str = ""
	size: NumberMultiplier = 2M
	kind?: "Superhero" | "War" | "Unknown"
	unknown1?: int | str = None
	unknown2?: any = None
```
will fix it